x-pack/plugins/security_solution_serverless/server/cloud_security/containters_defend.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import type { AggregationsAggregate, SearchResponse } from '@elastic/elasticsearch/lib/api/types';
+import { CLOUD_DEFEND, CLOUD_SECURITY_TASK_TYPE, CLOUD_DEFEND_HEARTBEAT_INDEX } from './constants';
+import type { Tier, UsageRecord } from '../types';
+import type { CloudSecurityMeteringCallbackInput } from './types';
+
+export interface CloudDefendHeartbeat {
+  '@timestamp': string;
+  'agent.id': string;
+  event: {
+    ingested: string;
+  };
+}
+
+const buildMeteringRecord = (
+  agentId: string,
+  timestampStr: string,
+  taskId: string,
+  tier: Tier,
+  projectId: string
+): UsageRecord => {
+  const timestamp = new Date(timestampStr);
+  timestamp.setMinutes(0);
+  timestamp.setSeconds(0);
+  timestamp.setMilliseconds(0);
+
+  const usageRecord = {
+    id: `container-defend-${agentId}-${timestamp.toISOString()}`,
+    usage_timestamp: timestampStr,
+    creation_timestamp: timestampStr,
+    usage: {
+      type: CLOUD_SECURITY_TASK_TYPE,
+      sub_type: CLOUD_DEFEND,
+      period_seconds: 3600,
+      quantity: 1,
+    },
+    source: {
+      id: taskId,
+      instance_group_id: projectId,
+      metadata: {
+        tier,
+      },
+    },
+  };
+
+  return usageRecord;
+};
+export const getHeartbeatRecords = async (
+  esClient: ElasticsearchClient,
+  searchFrom: Date
+): Promise<SearchResponse<CloudDefendHeartbeat, Record<string, AggregationsAggregate>>> => {
+  return await esClient.search<CloudDefendHeartbeat>(
+    {
+      index: CLOUD_DEFEND_HEARTBEAT_INDEX,
+      sort: 'event.ingested',
+      query: {
```

Comment:
We probably need an explicit `size` param here otherwise it will only pull 10 docs per run and miss the other heartbeats.